src/parameters/Hyperparameters.jl

```diff
@@ -14,7 +14,7 @@ end
     Hyperparameters(;
         current_epoch::Int64 = nothing,
         current_minibatch::Int64 = nothing,
-        loss_history::Vector{Float64} = Vector{Float64}[],
+        loss_history::Vector{Float64} = zeros(Float64, 0),
```

Comment:
Why do you want to initialize it as zeros? I think the idea was to have an empty array and to append new loss values as the training advanced.